.github/ISSUE_TEMPLATE/design-doc.md

```diff
@@ -0,0 +1,86 @@
+---
+name: Design document
+about: Create a proposal for a feature which includes substantial architectural work.
+title: ''
+labels: design-doc, feature
+assignees: ''
+
+---
+# UCC Design Doc
+
+<!-- Template to use for proposing substantial changes to UCC: architecture changes, major refactoring, new modules, etc. -->
+
+## Motivation
+Why should we build this? What problems are we trying to address?
+
+
+## Goals
+What are in-scope goals for this project? These should be high-level, but specific enough to map onto sub-issues.
+
+
+## Non-Goals
+What is out-of-scope for this project? What possible tools are we NOT building and which possible use cases are we NOT covering?
+
+
```

Comment:
It might be redundant with Goals and User Experience, but potentially a section on success criteria to answer "How will we know this achieves the goals?".